common/src/com/thoughtworks/go/config/materials/Materials.java

```diff
@@ -140,9 +134,9 @@ public void cleanUp(File baseFolder, ConsoleOutputStreamConsumer consumer) {
         return allowed;
     }
 
-    boolean hasOneMaterialUseBaseFolder() {
+    public boolean hasOneMaterialUseBaseFolder() {
         for (Material material : this) {
-            if (material.getFolder() == null) {
+            if (material.getFolder() == null && material instanceof SvnMaterial) {
```

Comment:
Can we do something other than an instanceof check? The tests are not very useful. For instance, they don't say what the behavior is when there are multiple materials. What does hasOneMaterialUseBaseFolder mean? If there's a dependency material and a git material, what should the behavior be? What about dependency and svn?